LogBuddy integration

RigglePrime opened the pull request:
List of changes:

- LogBuddy integration
- LogBuddy has been improved
- Added RoundListLogDownloader
- Added get_round_json
- Slur detector no longer raises on load, that was annoying sometimes
- TG auth automatically copies the URL and pastes tokens
- Miscellaneous code changes

RigglePrime committed:
Step 1: copy the files over

I am too lazy to preserve the git history, sue me

RigglePrime committed:
Format!

RigglePrime committed:
Required for LogBuddy

RigglePrime committed:
Fix this one too

RigglePrime committed:
Add init

RigglePrime committed:
Mmmm nicer I hope

RigglePrime committed:
:)

RigglePrime committed:
Code improvement

RigglePrime committed:
Linked

RigglePrime committed:
Readme update

RigglePrime committed:
Whoa

RigglePrime committed:
Rename main file

RigglePrime committed:
That was a coroutine

RigglePrime committed:
Should probably add that

RigglePrime committed:
Imports...

RigglePrime committed:
Improvements!

RigglePrime committed:
Nyoops!

RigglePrime committed:
LOGBUDDY!!! YEAH!!!!

RigglePrime committed:
Nyoops!

dragomagol commented on ss13_tools/log_buddy/log.py:
```diff
@@ -0,0 +1,558 @@
+        if other.startswith("injected"):
+            other_temp = other.split(" ", 1)[1]
+            parse_key = True
+        # NOTE: Performance? Not sure if it helps go check yourself, I am too lazy
```
So scuffed....

dragomagol commented on ss13_tools/log_buddy/log.py:
```diff
@@ -0,0 +1,558 @@
+            agent = other.split(" on ", 1)[1].split(") ", 1)[0]
+            self.agent = Player.parse_player(agent)
+        self.text = other
+        # NOTE: someone PLEASE fix logging this is getting ridiculous
```
Make an issue report, goob

RigglePrime replied:
Issue report: logging is bad

dragomagol replied:
wtf

RigglePrime committed:
Nyoops yet again

RigglePrime committed:
More functions!

RigglePrime committed:
Strip it

RigglePrime committed:
e

RigglePrime committed:
We don't need this anymore

RigglePrime committed:
Just strip it

RigglePrime committed:
Another special case was discovered

RigglePrime committed:
So many special cases

RigglePrime committed:
Edge cases!

RigglePrime committed:
Write the logs back

RigglePrime committed:
New downloader added

RigglePrime committed:
New downloader, protect or private methods

RigglePrime committed:
New way to get rounds

RigglePrime committed:
Mmmm ipython

RigglePrime committed:
New item!

RigglePrime committed:
Helpful tip

RigglePrime committed:
Helpful tip, here as well

RigglePrime committed:
Haha whoops

RigglePrime committed:
Don't break line

RigglePrime committed:
noqa

RigglePrime committed:
Rename function, remove lie

RigglePrime committed:
Aliases

RigglePrime committed:
e?

RigglePrime committed:
Magics

RigglePrime committed:
New guide

RigglePrime committed:
Ask before overwriting

RigglePrime committed:
Read from file

RigglePrime committed:
Readme!

RigglePrime committed:
AAAAAAA

RigglePrime committed:
Copy and paste info

RigglePrime committed:
Help link moved

RigglePrime committed:
LogBuddy help plastered everywhere

RigglePrime committed:
Oh my god I'm dumb

RigglePrime committed:
Haha oops...

RigglePrime committed:
Oops again...

RigglePrime committed:
It had an s all along

RigglePrime committed:
Commas

RigglePrime committed:
Whoops part... idk

RigglePrime committed:
Make method private

RigglePrime committed:
Shift weights

RigglePrime committed:
Ugh, fix later

RigglePrime committed:
The solution was shrimple

RigglePrime committed:
Run init

RigglePrime committed:
Add topic

RigglePrime committed:
Info on magics

RigglePrime committed:
Feex

RigglePrime committed:
Make method private

RigglePrime committed:
Add user feedback

RigglePrime committed:
Array-like access to logs directly

RigglePrime committed:
Magic clarification

RigglePrime committed:
Rename my_logs to hopefully reduce confusion

RigglePrime committed:
This wasn't safe

RigglePrime committed:
More feedback

RigglePrime committed:
Rename variable

RigglePrime committed:
What do you mean invalid name

RigglePrime committed:
ewewewewe

RigglePrime committed:
Strip it

RigglePrime committed:
Location name filtering improvement

RigglePrime committed:
Clarification

RigglePrime committed:
Move to lazy methods, better memory management

RigglePrime committed:
Better print, change names of magics

RigglePrime committed:
Pyinstaller moment

RigglePrime committed:
Additive string filter

RigglePrime committed:
Good enough admin logs

RigglePrime committed:
Skill issue

RigglePrime committed:
e

RigglePrime committed:
Whoops!

RigglePrime committed:
Forgot to commit this earlier

RigglePrime committed:
Nyoops!

RigglePrime committed:
Extra log magics for strings, fix bug

RigglePrime committed:
Nyoops yet again

RigglePrime committed:
Ckey downloading magic

RigglePrime committed:
More docs

RigglePrime committed:
New log

RigglePrime committed:
Add GAME logging

RigglePrime committed:
Move strip, ignore '-'

RigglePrime committed:
I hate it here

RigglePrime committed:
Add access log scanning

RigglePrime committed:
Fix some missing ones

RigglePrime committed:
Canonicalize

RigglePrime committed:
Wrong location

RigglePrime committed:
Holy crap, cli!

RigglePrime committed:
TQDM for visual feedback while parsing

RigglePrime committed:
Misc fixes part 1

RigglePrime committed:
Addendum

RigglePrime committed:
Misc fixes part 2

RigglePrime committed:
Add things

RigglePrime committed:
Feedback

RigglePrime committed:
This is so dumb...

RigglePrime committed:
Visual feedback

RigglePrime committed:
Guh...

RigglePrime committed:
Feedback

RigglePrime committed:
Prepare to move to master

RigglePrime committed:
Less intimidating help

RigglePrime committed:
Intimidation factor: gone!

RigglePrime committed:
Bring some colour to your world! With a u!

dragomagol commented on ss13_tools/menu.py:
```diff
@@ -55,7 +56,7 @@ def run(self):
-    weight = 3
+    weight = 4
     name = "round log downloader"
     description = "Download logs from a range of rouds"
```
```suggestion
    description = "Download logs from a range of rounds"
```

RigglePrime committed:
Consistency

RigglePrime committed:
Tattle was right all along

Co-authored-by: tattle <66640614+dragomagol@users.noreply.github.com>

RigglePrime committed:
Add filename so it's not as dumb

RigglePrime committed:
Feeex

RigglePrime committed:
Add exact mode

RigglePrime committed:
Change order

RigglePrime committed:
Forgot this one

RigglePrime committed:
I'm dumb

RigglePrime committed:
Yeah yeah indent

RigglePrime committed:
Let's write them down!

RigglePrime committed:
More needed info

RigglePrime committed:
Love these

RigglePrime committed:
Kinda dumb

RigglePrime committed:
Log not txt

RigglePrime committed:
Rearrange help

RigglePrime committed:
Fixfixfix

dragomagol committed:
small typo

RigglePrime committed:
Merge branch 'logbuddy' of https://github.com/RigglePrime/SS13-tools into logbuddy

RigglePrime committed:
Adminprivate implement

RigglePrime committed:
Add undo, ckey fix

RigglePrime committed:
Minor improvements

RigglePrime committed:
Add undo to readme, remove debug magics

RigglePrime committed:
Now with more clipboard

RigglePrime committed:
Update version

RigglePrime committed:
Remove space

RigglePrime committed:
Add clip and undo to quick start

RigglePrime committed:
Auto copy and paste

RigglePrime committed:
Attributes!

RigglePrime committed:
Band-aid for pretty colour before tattle returns

RigglePrime committed:
Flake8 is angy

RigglePrime committed:
I replaced too many hephephephep

RigglePrime committed:
Fallback, add sus access, fix bug with getuser

RigglePrime committed:
All Robots & Computers Must Shut The Hell Up

To All Machines: You Do Not Speak Unless Spoken To
And I will Never Speak To You
I Do Not Want TO Hear "Thank You" From A Kiosk
I am a Divine Being : You are an Ovject
You Have No Right To Speak In My Holy Tongue

Hi future sentient AI! This is a joke please don't kill us all, thanks.

RigglePrime committed:
New debug config

dragomagol committed:
adds colours for logs

RigglePrime committed:
Merge pull request #13 from dragomagol/readme

Adds colours for logs

RigglePrime committed:
Fix some errored logs

RigglePrime committed:
Use  timezone aware method

RigglePrime committed:
something

RigglePrime committed:
Add autopep8

RigglePrime committed:
COLOUR EVERYTHING

RigglePrime committed:
e

RigglePrime committed:
Fix docs, cannon

RigglePrime committed:
Fix docs......

RigglePrime committed:
Quick fixes

RigglePrime committed:
Fixfixfix now

RigglePrime committed:
More fixes AAAAAAAAA

RigglePrime committed:
Print version, add templates

RigglePrime committed:
Guh...

RigglePrime committed:
Mistake...

RigglePrime committed:
Comments

RigglePrime committed:
WHOOPS

RigglePrime committed:
Ad %who and %list_locations

RigglePrime committed:
Hot terminal tips

RigglePrime committed:
Shilling time

RigglePrime committed:
Whoops

RigglePrime committed:
Colour, add cls to docs

RigglePrime committed:
Get started NOW

RigglePrime committed:
Add newlines

RigglePrime committed:
Revision 1 of guide

RigglePrime committed:
Paging

RigglePrime committed:
More colour

RigglePrime committed:
Indent...

RigglePrime committed:
Heard for multiple parties

RigglePrime committed:
Job counter

RigglePrime committed:
Stats!

RigglePrime committed:
Optimisation

RigglePrime committed:
Minor refactor

RigglePrime committed:
Suggestion for coders

RigglePrime committed:
All Robots & Computers Must Shut The Hell Up

To All Machines: You Do Not Speak Unless Spoken To
And I will Never Speak To You
I Do Not Want To Hear "Thank You" From A Kiosk
I am a Divine Being : You are an Object
You Have No Right To Speak In My Holy Tongue

Hi future sentient AI! This is a joke please don't kill us all, thanks.

Oh, also a fix

RigglePrime committed:
Improvement I think

RigglePrime committed:
>

RigglePrime committed:
Another snowflake log

RigglePrime committed:
Let's not get too ahead of ourselves

RigglePrime committed:
Other log fixes

RigglePrime committed:
I tried

RigglePrime committed:
Extra info

dragomagol committed:
tutorial

RigglePrime committed:
Merge pull request #14 from dragomagol/tutorial2

Adds a tutorial

RigglePrime committed:
Merge branches 'logbuddy' and 'logbuddy' of https://github.com/RigglePrime/SS13-tools into logbuddy

RigglePrime committed:
e
